lib/WiFiNative/src/WiFiClient.h

```diff
@@ -0,0 +1,145 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Wifi Client for Native platform. Emulation of Arduino WiFiClient.
+ * @author Norbert Schulz <github@schulznorbert.de>
+ *
+ * @addtogroup HALSim
+ *
+ * @{
+ */
+#ifndef WIFI_CLIENT_H
+#define WIFI_CLIENT_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+
+#include <Arduino.h>
+#include <IPAddress.h>
+
+#include <poll.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * WiFiClient client class compatible with Arduino. 
+ * 
+ * Only a subset of API functions is supported, based on the actual demand.
+ */
+class WiFiClient
+{
+    public:
+
+    /**
+     * Default constructor.
+     */
+    WiFiClient() :
+        m_poll {SOCK_INVAL, POLLIN, 0}
+    {
+    }
+
+    /**
+     * Default Destructor.
+     */
+    virtual ~WiFiClient()
+    {
+    }
+
+    /**
+     * Establish a connection.
+     * 
+     * @param[in] addr Port number.
+     * @param[in] port Port number.
+     * @return If successful returns 1, otherwise 0.
+     */
+    uint8_t connect(const IPAddress& addr, uint16_t port);
+
+    /**
+     * Indicate if client is connected.
+     * 
+     * @return If successful returns 1, otherwise 0.
+     */
+    uint8_t connected()
+    {
+        return (SOCK_INVAL != m_poll.fd) ? 1U : 0U;
+    }
+
+    /**
+     * Stop the WiF Client.
+     */
+    void stop();
+
+    /**
+     * Write bytes to stream.
```

Comment:
Shouldn't we add that this is a blocking write?

Reply:
You found a bug. O_NONBLOCK is used, but write doesn't handle it. Updating and documenting write to behave like this:

   /**
     * Write bytes to stream.
     *
     * The write uses non blocking socket mode, but retries on a
     * EWOULDBLOCK results. The constants SOCK_WRITE_RETRY and
     * SOCK_WRITE_TMO_US define how often to retry and the 
     * time delay in between.
     * ...